src/DateTimeInput/index.jsx

```diff
@@ -396,22 +438,20 @@ export default class DateTimeInput extends Component
         } );
 
         return _.chunk( days, 7 );
-    }
+    };
 
-    monthMatrix()
+    const monthMatrix = () =>
```

Comment:
This function should probably be extracted outside of the main DateTimeInput function

Reply:
Same for `dayMatrix`

Reply:
As discussed, this would complicate implementation of both functions, as well as `isUnitSelectable` that they're using, so we decided to keep this inside component.